presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,290 @@
+import static io.prestosql.spi.type.TinyintType.TINYINT;
+import static io.prestosql.spi.type.Varchars.isVarcharType;
+import static java.lang.Integer.parseInt;
+import static java.lang.String.format;
+
+public class RawRowEncoder
+        extends AbstractRowEncoder
+{
+    private enum FieldType
+    {
+        BYTE(Byte.SIZE),
+        SHORT(Short.SIZE),
+        INT(Integer.SIZE),
+        LONG(Long.SIZE),
+        FLOAT(Float.SIZE),
+        DOUBLE(Double.SIZE);
+
+        private final int size;
+
+        FieldType(int bitSize)
+        {
+            this.size = bitSize / 8;
+        }
+
+        public int getSize()
+        {
+            return size;
+        }
+    }
+
+    private static final Pattern MAPPING_PATTERN = Pattern.compile("(\\d+)(?::(\\d+))?");
+    private static final Set<Type> SUPPORTED_PRIMITIVE_TYPES = ImmutableSet.of(
+            BIGINT, INTEGER, SMALLINT, TINYINT, DOUBLE, REAL, BOOLEAN);
+
+    public static final String NAME = "raw";
+
+    private final ByteBuffer buffer;
+    private int[] valueLengths;
+    private int currentBufferPosition;
+
+    public RawRowEncoder(ConnectorSession session, List<EncoderColumnHandle> columnHandles)
+    {
+        super(session, columnHandles);
+
+        int capacity = 0;
+        for (int i = 0; i < columnHandles.size(); i++) {
+            capacity += valueLengths[i];
+        }
+
+        this.buffer = ByteBuffer.allocate(capacity);
+        this.currentBufferPosition = 0;
+    }
+
+    @Override
+    protected void validateColumns(List<EncoderColumnHandle> columnHandles)
+    {
+        int length;
+        EncoderColumnHandle columnHandle;
+        valueLengths = new int[columnHandles.size()];
+
+        for (int i = 0; i < columnHandles.size(); i++) {
+            columnHandle = columnHandles.get(i);
+            checkArgument(columnHandle.getFormatHint() == null, "unexpected format hint '%s' defined for column '%s'", columnHandle.getFormatHint(), columnHandle.getName());
+
+            FieldType fieldType = parseFieldType(columnHandle);
+            checkFieldType(columnHandle, fieldType);
+
+            length = parseMapping(columnHandle, fieldType);
+            valueLengths[i] = length;
+
+            checkArgument(isSupportedType(columnHandle.getType()), "Unsupported column type '%s' for column '%s'", columnHandle.getType().getDisplayName(), columnHandle.getName());
+        }
+    }
+
+    private FieldType parseFieldType(EncoderColumnHandle columnHandle)
+    {
+        FieldType fieldType;
+        try {
+            fieldType = Optional.ofNullable(columnHandle.getDataFormat())
+                    .map(dataFormat -> FieldType.valueOf(dataFormat.toUpperCase(Locale.ENGLISH)))
+                    .orElse(FieldType.BYTE);
+        }
+        catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(format("invalid dataFormat '%s' for column '%s'", columnHandle.getDataFormat(), columnHandle.getName()));
```

Comment:
wrap original exception in thrown one.

Reply:
Or maybe we can just rethrow original exception. Would that be enough? Are we adding extra information to message by wrapping here?

Reply:
Speaking from experience it is pretty helpful to know which column is problematic hahah